routing/probability_bimodal.go

```diff
@@ -301,6 +301,28 @@ func (p *BimodalEstimator) calculateProbability(directProbability float64,
 		return directProbability
 	}
 
+	// If we have up-to-date information about the channel we want to use,
+	// i.e. the info stems from results no longer ago than the decay time,
+	// we will only use the direct probability. This is needed in order to
+	// avoid that other results of the same node will distort the calculated
+	// probability even if we have accurate direct information. This also
+	// helps to dip the probability below the min probability in case of
+	// failures, to start the splitting process.
+	directResult, ok := results[toNode]
+	if ok {
+		latest := directResult.SuccessTime
+		if directResult.FailTime.After(latest) {
+			latest = directResult.FailTime
+		}
+
+		if now.Sub(latest) < p.BimodalDecayTime {
```

Comment:
isn't this to aggressive? Should we have a different constant only for this?

Reply:
actually a second thought is we don't need to compare these two values right? we don't care if `directProbability` is greater or smaller than the new calculated one below, but we always prefer the more recent one.

Reply:
We always prefer `directProbability`, if it's recent. In this case we also don't care about the mixed probability below, as that could skew the picture. `directProbability` can be higher (in the success case) or lower (in the failure case) compared to the mixed probability.